Bugzilla/Quantum/OAuth2/Clients.pm

```diff
@@ -0,0 +1,210 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Quantum::OAuth2::Clients;
+use Mojo::Base 'Mojolicious::Controller';
+
+use 5.10.1;
+use List::Util qw(first);
+use Moo;
+
+use Bugzilla;
+use Bugzilla::Error;
+use Bugzilla::Token;
+
+
+# Show list of clients
+sub list {
+  my ($self) = @_;
+  my $clients = Bugzilla->dbh->selectall_arrayref('SELECT * FROM oauth2_client',
+    {Slice => {}});
+  $self->stash(clients => $clients);
+  return $self->render(template => 'admin/oauth/list', handler => 'bugzilla');
+}
+
+# Create new client
+sub create {
+  my ($self) = @_;
+  my $dbh    = Bugzilla->dbh;
+  my $vars   = {};
+
+  if ($self->req->method ne 'POST') {
+    $vars->{token} = issue_session_token('create_oauth_client');
+    $vars->{scopes}
+      = $dbh->selectall_arrayref('SELECT * FROM oauth2_scope', {Slice => {}});
+    $self->stash(%{$vars});
+    return $self->render(template => 'admin/oauth/create',
+      handler => 'bugzilla');
+  }
+
+  $dbh->bz_start_transaction;
+
+  my $description = $self->param('description');
+  my $id          = $self->param('id');
```

Comment:
Neither the client id or the client secret should be a parameter. Make them randomly generated please. Also remove them from the editing page too.
Only the description is controlled by the user.